modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -327,11 +331,55 @@ public void start(@Nullable String cfg) {
                 doStartComponent(name, startedComponents, component);
             }
 
+            CompletableFuture<Void> upToDateMetastorageRevisionFut = new CompletableFuture<>();
+
+            metaStorageMgr.listen(MetastorageEvent.REVISION_APPLIED, new EventListener<MetastorageEventParameters>() {
```

Comment:
I'd recommend moving the listener body to separate method

Reply:
Moved.